scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java

```diff
@@ -797,72 +885,102 @@ private ClickHandler clickHandlerForCheckButton() {
-                                    if (((JSONBoolean) obj.get("valid")).booleanValue()) {
-                                        if (obj.containsKey("updatedVariables")) {
-                                            updateVariables(obj.get("updatedVariables"));
-                                            redrawVariables();
-                                        }
-                                        GWT.log("Job validated");
-                                        displayInfoMessage("Job is valid");
-                                    } else if (obj.containsKey("errorMessage")) {
-                                        String errorMessage = obj.get("errorMessage").toString();
-                                        if (errorMessage.contains("JobValidationException")) {
-                                            errorMessage = errorMessage.substring(errorMessage.indexOf(":") + 2);
-                                        }
-                                        displayErrorMessage(errorMessage);
+                            handleVariables(obj);
+                        } catch (Throwable t) {
+                            handleParsingError(result);
+                        }
+                    }
+
+                    private void handleParsingError(String result) {
+                        // JSON parsing error workaround to force extract error message
+                        MatchResult errorMessageMatcher = RegExp.compile(ERROR_MESSAGE_REGEX).exec(result);
+                        if (errorMessageMatcher != null) {
+                            GWT.log(errorMessageMatcher.getGroup(1));
+                            displayErrorMessage(errorMessageMatcher.getGroup(1));
+                        } else {
+                            GWT.log(JSON_ERROR);
+                            displayErrorMessage(JSON_ERROR);
+                        }
+                    }
+
+                    private void handleVariables(JSONObject obj) {
+                        if (obj != null) {
+                            if (obj.containsKey("valid")) {
+                                if (((JSONBoolean) obj.get("valid")).booleanValue()) {
+                                    if (obj.containsKey("updatedVariables")) {
+                                        updateVariables(obj.get("updatedVariables"));
+                                        redrawVariables(job);
+                                    }
+                                    GWT.log("Job validated");
+                                    displayInfoMessage("Job is valid");
+                                } else if (obj.containsKey(ERROR_MESSAGE)) {
+                                    String errorMessage = obj.get(ERROR_MESSAGE).toString();
+                                    if (errorMessage.contains("JobValidationException")) {
+                                        errorMessage = errorMessage.substring(errorMessage.indexOf(":") + 2);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Put single-quotes around ':' to use the faster "indexOf(char)" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3027)